core/__tests__/models/destination/plugins/processExportedProfiles.ts

```diff
@@ -0,0 +1,720 @@
+
+      // Failed exports get their error set
+      await export1.reload();
+      expect(export1.profileId).toBe(profile1.id);
+      expect(export1.state).toBe("pending"); // kicked back to pending state
+      expect(export1.completedAt).toBeFalsy();
+      expect(export1.errorMessage).toBe("oh no!");
+      expect(export1.errorLevel).toBe("error");
+      expect(export1.exportProcessorId).toBeNull(); // cleared
+
+      await export2.reload();
+      expect(export2.profileId).toBe(profile2.id);
+      expect(export2.state).toBe("failed"); // marked as failed
+      expect(export2.completedAt).toBeFalsy();
+      expect(export2.errorMessage).toBe("info error!");
+      expect(export2.errorLevel).toBe("info");
+      expect(export2.exportProcessorId).toBe(exportProcessor.id);
+
+      // Successful exports will be marked as complete
+      await export3.reload();
+      expect(export3.profileId).toBe(profile3.id);
+      expect(export3.state).toBe("complete");
+      expect(export3.completedAt).toBeTruthy();
+      expect(export3.errorMessage).toBeNull();
+      expect(export3.exportProcessorId).toBe(exportProcessor.id);
+
+      await export1.destroy();
+      await profile1.destroy();
+      await profile2.destroy();
+      await profile3.destroy();
+    });
+
+    test("the processExportedProfiles method can return processExports to have them reprocessed", async () => {
+      const exportProcessor = await helper.factories.exportProcessor(
+        destination,
+        {
+          processAt: new Date(0),
+          remoteKey: "my-remote-key",
+        }
+      );
+      const profile1 = await helper.factories.profile();
+      const profile2 = await helper.factories.profile();
+      const profile3 = await helper.factories.profile();
+      const export1 = await helper.factories.export(profile1, destination, {
+        startedAt: new Date(),
+        state: "processing",
+        exportProcessorId: exportProcessor.id,
+      });
+      const export2 = await helper.factories.export(profile2, destination, {
+        startedAt: new Date(),
+        state: "processing",
+        exportProcessorId: exportProcessor.id,
+      });
+      const export3 = await helper.factories.export(profile3, destination, {
+        startedAt: new Date(),
+        state: "processing",
+        exportProcessorId: exportProcessor.id,
+      });
+
+      let foundTasks = await specHelper.findEnqueuedTasks("export:process");
+      expect(foundTasks.length).toBe(0);
+
+      await specHelper.runTask("export:enqueueProcessors", {});
+
+      foundTasks = await specHelper.findEnqueuedTasks("export:process");
+      expect(foundTasks.length).toBe(1);
+
+      processProfilesResponse = {
+        success: true,
+        processExports: {
+          profileIds: [profile1.id, profile2.id], // profile3 was successful
+          processDelay: 1000,
+        },
+        errors: null,
+        retryDelay: null,
+      };
```

Comment:
This is an interesting use-case.  I don't know if a destination would say that some of the batch of profiles are processing and others aren't, but I guess the functionality is nice to have.

Reply:
yeah, I didn't see an immediate need for the use case but allowing it was straightforward enough